src/test/java/homework16/ProductTest.java

```diff
@@ -0,0 +1,92 @@
+package homework16;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.List;
+
+import static homework16.Product.getHighPriceBook;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ProductTest {
+
+    @Test
+    void shouldGetHighPriceBook() {
+        List<Product> products = new ArrayList<Product>();
+        products.add(new Product(ProductType.BOOK, 75, true, LocalDate.now()));
+        products.add(new Product(ProductType.MILK, 50, false, LocalDate.now()));
+        products.add(new Product(ProductType.BOOK, 300, false, LocalDate.now()));
+        products.add(new Product(ProductType.BOOK, 400, true, LocalDate.now()));
+        List<Product> expected = new ArrayList<>();
+        expected.add(new Product(ProductType.BOOK, 300, false, LocalDate.now()));
+        expected.add(new Product(ProductType.BOOK, 400, true, LocalDate.now()));
+        assertEquals(getHighPriceBook(products).toString(), expected.toString());
```

Comment:
строкове представлення обʼєкта не має ніколи порівнюватись, що зрозуміти що обʼєкти однакові. 
Пригадуй коли обʼєкти однакові? У списків з цим все гаразд, але проблема в тому, що цього немає в твоїй реалізації Product. 

Має так працювати: 
```suggestion
        assertEquals(getHighPriceBook(products), expected);
```
В усіх тестах